dali/ft/filecopy.cpp

```diff
@@ -1736,8 +1736,12 @@ void FileSprayer::locateJsonHeader(IFileIO * io, unsigned headerSize, offset_t &
     Owned<IFileIOStream> in = createIOStream(io);
 
     JsonSplitter jsplitter(srcFormat, *in);
+    if (headerSize > 0)
+        in->seek(headerSize, IFSbegin); // Skip BOM
     headerLength = jsplitter.getHeaderLength();
     footerLength = jsplitter.getFooterLength();
+    if (headerSize > 0)
```

Comment:
This should have a comment to explain why the footer is adjusted with the headerSize.  It looks like it is depending on an implementation detail.  I suspect headerSize should be passed to JsonSplitter instead and the adjustments made within that class.